docs/ansible/coding-conventions.md

```diff
@@ -0,0 +1,838 @@
+
+```yaml
+---
+# file: webservers.yml
+- hosts: webservers
+  roles:
+    - common
+    - webtier
+```
+
+The idea here is that we can choose to configure our whole infrastructure by “running” site.yml or we could just choose to run a subset by running webservers.yml.
+This is analogous to the “–limit” parameter to ansible but a little more explicit:
+
+```yaml
+ansible-playbook site.yml --limit webservers
+ansible-playbook webservers.yml
+```
+
+
+## Bundling Ansible Modules With Playbooks
+
+If a playbook has a ./library directory relative to its YAML file, this directory can be used to add ansible modules that will automatically be in the ansible module path.
+This is a great way to keep modules that go with a playbook together.
+This is shown in the directory structure example at the start of this section.
+
+## Playbooks optimization
+
+### Import vs Include
+
+Alway try to break up playbooks into smaller logically separated parts.
+
+Includes and imports allow users to do this, which can be used across multiple parent playbooks or even multiple times within the same Playbook.
+A major difference between `import` and `include` tasks:
+
+- `import` tasks will be parsed at the beginning when you run your playbook
+(If you use any `import*` Task (`import_playbook`, `import_tasks`, etc.), it will be static.)
+- `include` tasks will be parsed at the moment Ansible hits them
+(If you use any **include*** Task (**include_tasks**, **include_role**, etc.), it will be dynamic.)
+
+  
+**Warning:** The bare `include` task (which was used for both Task files and Playbook-level includes) is still available, however it is now considered deprecated.
+
+#### Tradeoffs and Pitfalls Between Includes and Imports
+* Using `include*` vs. `import*` has some advantages as well as some tradeoffs which users should consider when choosing to use each:
+
+* The primary advantage of using `include*` statements is looping. When a loop is used with an include, the included tasks or role will be executed once for each item in the loop.
+
+* Using `include*` does have some limitations when compared to `import*` statements:
+
+* Tags which only exist inside a dynamic include will not show up in `--list-tags` output.
+
+* Tasks which only exist inside a dynamic include will not show up in `--list-tasks` output.
+
+* You cannot use `notify` to trigger a handler name which comes from inside a dynamic include (see note below).
+
+* You cannot use `--start-at-task` to begin execution at a task inside a dynamic include.
+
+* Using `import*` can also have some limitations when compared to dynamic includes:
+
+* As noted above, loops cannot be used with imports at all.
+
+* When using variables for the target file or role name, variables from inventory sources (**host**/**group** vars, etc.) cannot be used.
+
+* Handlers using **import*** will not be triggered when notified by their name, as importing overwrites the handler’s named task with the imported task list.
```

Comment:
fix list layout because it has several levels (according to the content), but yours is single

Reply:
If you mean the directory tree - i just copied it from OrchidE example. 
I also completely remade this block, so i think currrently everything is ok here